cli/src/pcluster/utils.py

```diff
@@ -1247,3 +1247,25 @@ def get_instance_gpus(instance_type, instance_info=None):
     gpus = sum([gpus.get("Count") for gpus in gpu_info.get("Gpus")]) if gpu_info else 0
 
     return gpus
+
+
+def get_default_instance_type():
+    """If current region support free tier, return the free tier instance type. Otherwise, return t3.micro ."""
+    if not hasattr(get_default_instance_type, "cache"):
+        get_default_instance_type.cache = {}
+    cache = get_default_instance_type.cache
+    region = os.environ.get("AWS_DEFAULT_REGION")
+    if region not in cache:
+        ec2_client = boto3.client("ec2")
+        paginator = ec2_client.get_paginator("describe_instance_types")
```

Comment:
we have a `paginate_boto3` utility method in the `utils.py` module. We can reuse it.